### PR TITLE
Add option to skip docker pull in update-felix make target.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Makes it easier to do builds with a locally-built felix image.

## Todos

- [x] Tests
- [x] Documentation
- [x] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: calico_node/Makefile [hunk 1]
```diff
@@ -195,7 +198,7 @@ $(NODE_CONTAINER_BIN_DIR)/calico-felix update-felix:
 	-docker rm -f calico-felix
 	# Latest felix binaries are stored in automated builds of calico/felix.
 	# To get them, we create (but don't start) a container from that image.
-	docker pull $(FELIX_CONTAINER_NAME)
+	if $(PULL_FELIX); then docker pull $(FELIX_CONTAINER_NAME); fi
```
Comment (neiljerram) [inline calico_node/Makefile]: This is fine - but FWIW I like the shorter:

```
$(PULL_FELIX) || docker pull $(FELIX_CONTAINER_NAME)
```

Comment (fasaxc) [reply calico_node/Makefile]: Except that'd be backwards!  It'd pull felix only if the flag was false.

Comment (neiljerram) [reply calico_node/Makefile]: D'oh, yes.  Would need to `&&` instead of `||`.

Comment (fasaxc) [reply calico_node/Makefile]: But then `false && <something>` has a false return code :-(  I think it needs to be `! $(PULL_FELIX) || ...`, but that's starting to get a bit convoluted.

Comment (neiljerram) [reply calico_node/Makefile]: Indeed, it's a lot more fiddly than I thought!  Bearing that in mind, I'm happy that my suggestion doesn't help.

**Pull request merged**